src/java/picard/analysis/CollectMultipleMetrics.java

```diff
@@ -368,7 +370,7 @@ public static void main(final String[] args) {
      * Use this method when invoking CollectMultipleMetrics programmatically to run programs other than the ones
      * available via enum.  This must be called before doWork().
      */
-    public void setProgramsToRun(final List<ProgramInterface> programsToRun) {
+    public void setProgramsToRun(final Set<ProgramInterface> programsToRun) {
```

Comment:
It would be nicer to change this to `Collection<ProgramInterface>` since that would be backwards compatible, and then implement as `this.programToRun.clear(); this.programsToRun.add(programsToRun)` so that a) it works and b) the caller can't silently modify the collection be retaining a reference.